pkg/data/helpers.go

```diff
@@ -9,18 +9,28 @@ import (
 	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
 )
 
-func getFiltersFromStruct(i interface{}, keyName *string) (*expression.KeyConditionBuilder, *expression.ConditionBuilder) {
+type stringer interface {
+	String() string
+}
+
+func getFiltersFromStruct(input interface{}, keyName *string) (*expression.KeyConditionBuilder, *expression.ConditionBuilder) {
 	var cb *expression.ConditionBuilder
 	var kb *expression.KeyConditionBuilder
-	v := reflect.ValueOf(i).Elem()
+	var dValue interface{}
+	v := reflect.ValueOf(input).Elem()
 	for i := 0; i < v.NumField(); i++ {
 		dField := strings.Split(v.Type().Field(i).Tag.Get("dynamodbav"), ",")[0]
 		if dField != "-" {
 			value := v.Field(i).Interface()
 			if !reflect.ValueOf(value).IsNil() {
-				switch v.Field(i).Kind() {
+				t := v.Field(i)
+				switch t.Kind() {
 				case reflect.Ptr:
-					dValue := reflect.Indirect(v.Field(i)).Interface()
+					if u, ok := t.Interface().(stringer); ok {
```

Comment:
Looks like when the build function takes a value it doesn't use the dynamodbattribute builderrs like in other areas of the code.  This means I need to manually convert custom types to the type that I want.  I decided to do that by looking for a String function and using that to build the value if it exists.